File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/RegionalGeocoder.java
package com.skedgo.tripkit.ui.geocoding;

import com.skedgo.tripkit.common.model.region.Region;
import com.skedgo.tripkit.ui.TripKitUI;

import java.util.List;

public class RegionalGeocoder extends Geocoder {
    public RegionalGeocoder() {
    }

    @Override
    public String getServiceUrl() {
        double latitude = getNearLatitude();
        double longitude = getNearLongitude();
        final Region r = TripKitUI.getInstance().regionService()
            .getRegionByLocationAsync(latitude, longitude)
            .blockingFirst();
        final List<String> urls = r.getURLs();
        return urls.get(0);
    }
}

File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/RegionalGeocoder.kt
package com.skedgo.tripkit.ui.geocoding

import com.skedgo.tripkit.ui.TripKitUI

open class RegionalGeocoder : Geocoder() {
    override val serviceUrl: String
        get() {
            val latitude = nearLatitude
            val longitude = nearLongitude
            // This can trigger a regions refresh (network). Never let a DNS outage crash the app.
            val region = TripKitUI.getInstance()
                .regionService()
                .getRegionByLocationAsync(latitude, longitude)
                .onErrorResumeNext(io.reactivex.Observable.empty())
                .firstElement()
                .blockingGet()
            val urls: List<String> = region.getURLs().orEmpty()
            return urls.firstOrNull().orEmpty()
        }
}
